pkg/operation/botanist/component/etcd/etcd.go

```diff
@@ -498,20 +498,18 @@ func (e *etcd) Deploy(ctx context.Context) error {
 			Quota:                   &quota,
 		}
 
-		if e.hasHAControlPlane() {
```

Comment:
Why was this removed?

Reply:
The reasoning is that if a non-HA shoot cluster (new) in future wishes to upgrade to a have a failure tolerance of `node` or `zone` then it will require 2 restarts of the existing etcd member before it can be scaled up to 3 nodes. If we enable it by default for all then we can completely avoid 2 restarts.

Reply:
made the change. PTAL

Reply:
As discussed with @rfranzke and @unmarshall, we plan to rollout PeerTLS to all clusters because it brings several benefits:
- Zero downtime upgrades to HA
- Consistent landscape
- Simpler code

We anyway rollout a new `etcd-backup-restore` version in the next Gardener release which will result in etcd restarts. So it'd be perfect if we combine this with the PeerTLS activation.

Reply:
Ok done the changes. PTAL
Cleaned up the commits and squashed it into 1 commit.